python_modules/dagster/dagster/_utils/__init__.py

```diff
@@ -166,7 +167,7 @@ def file_relative_path(dunderfile: str, relative_path: str) -> str:
     check.str_param(dunderfile, "dunderfile")
     check.str_param(relative_path, "relative_path")
 
-    return os.path.join(os.path.dirname(dunderfile), relative_path)
+    return os.fspath(Path(dunderfile, "..", relative_path).resolve())
```

Comment:
is using `.parent` more canonical than `..`?

Reply:
It supposedly is, but it looks incredibly ugly if you're trying to use it multiple times (e.g. `Path("my/path").parent.parent.parent`). So I end up not using it in favor of `..`.